web/modules/custom/bnf/src/Services/BnfImporter.php

```diff
@@ -20,46 +26,235 @@
  */
 class BnfImporter {
 
+  const ALLOWED_PARAGRAPHS = [
+    'ParagraphTextBody' => 'text_body',
+    'ParagraphAccordion' => 'accordion',
+  ];
+
   /**
    * Constructor.
    */
   public function __construct(
     protected ClientInterface $httpClient,
+    protected EntityFieldManagerInterface $entityFieldManager,
     protected EntityTypeManagerInterface $entityTypeManager,
     protected TranslationInterface $translation,
     protected LoggerInterface $logger,
   ) {}
 
   /**
-   * Importing a node from a GraphQL source endpoint.
+   * Loading the columns of a field, that we use to ask GraphQL for data.
+   *
+   * E.g. a WYSIWYG field will have both a "value" and a "format" that we want
+   * to pull out.
+   *
+   * @return mixed[]
+   *   Return an array of fields, along with their column keys.
    */
-  public function importNode(string $uuid, string $endpointUrl, string $nodeType = 'article'): void {
-    $nodeStorage = $this->entityTypeManager->getStorage('node');
+  protected function getFieldColumns(string $entityType, string $bundle): array {
+    $values = [];
+    $fields = [];
+    $fieldDefinitions = $this->entityFieldManager->getFieldDefinitions($entityType, $bundle);
 
-    $existingNodes =
-      $nodeStorage->loadByProperties(['uuid' => $uuid]);
+    foreach ($fieldDefinitions as $fieldKey => $fieldDefinition) {
+      if ($fieldDefinition instanceof FieldConfig) {
+        $fields[] = $fieldKey;
+      }
+    }
 
-    if (!empty($existingNodes)) {
-      $this->logger->error(
-        'Cannot import @type @uuid from @url - Node already exists.',
-        ['@type' => $nodeType, '@uuid' => $uuid, '@url' => $endpointUrl]
-      );
+    foreach ($fields as $fieldKey) {
+      $field = $this->entityTypeManager->getStorage('field_storage_config')->load("$entityType.$fieldKey");
 
-      throw new AlreadyExistsException('Cannot import node - already exists.');
+      if ($field instanceof FieldStorageConfig) {
+        $values[$fieldKey] = array_keys($field->getColumns());
+      }
     }
 
-    // Example of GraphQL query: "nodeArticle".
-    $queryName = 'node' . ucfirst($nodeType);
+    return $values;
+  }
 
-    // For now, we only support the title of the nodes.
+  /**
+   * Builds the query used to get data from the source.
+   */
+  public function getQuery(string $uuid, string $queryName): string {
+    // Start building the GraphQL query.
     $query = <<<GRAPHQL
     query {
       $queryName(id: "$uuid") {
         title
+        paragraphs {
+
+    GRAPHQL;
+
+    // Loop through allowed paragraphs and add their structures.
+    foreach (self::ALLOWED_PARAGRAPHS as $graphBundle => $drupalBundle) {
```

Comment:
I'd suggest to fold this out in a helper method that deals with the "inside" of the paragraphs section.

I think at the end we'll end up with a `getQuery` that just calls `getFields` which recursively generates the needed GraphQL.